File: 1]. DSA + CP/1]. DSA/1]. Data Structures/06]. Linked List/C++/_01)_Insert_At_Beginning_Of_LinkedList.cpp
#include <bits/stdc++.h> 
using namespace std; 

struct Node{
    int data;
    Node* next;
    Node(int x){
        data=x;
        next=NULL;
    }
};

Node *insertBegin(Node *head,int x){
    Node *temp=new Node(x);
    temp->next=head;
    return temp;
    
}

void printlist(Node *head){
        Node *curr=head;
        while(curr!=NULL){
        cout<<curr->data<<" ";
        curr=curr->next;
        }
}
int main() 
{ 
	Node *head=NULL;
	head=insertBegin(head,30);
	head=insertBegin(head,20);
	head=insertBegin(head,10);
	printlist(head);
	return 0;
} 

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/06]. Linked List/Java/_01)_Insert_At_Beginning _Of_LinkedList.java
import java.util.*;
import java.io.*;
import java.lang.*;

class Node{
        int data;
        Node next;
        Node(int x){
            data=x;
            next=null;
        }
    }
  
class Test { 
    
    
    static Node insertBegin(Node head, int x){
        Node temp=new Node(x);
        temp.next=head;
        return temp;
    }
    
    public static void main(String args[]) 
    { 
        Node head=null;
	    head=insertBegin(head,30);
	    head=insertBegin(head,20);
	    head=insertBegin(head,10);
	    printlist(head);
    	
    } 
    
    public static void printlist(Node head){
        Node curr=head;
        while(curr!=null){
        System.out.print(curr.data+" ");
        curr=curr.next;
    }
    }
} 
